### DISCUSSION TITLE
Pull request info missing from `repos/OWNER/REPO/actions/runs` (when run was trigger by a PR)

### DISCUSSION THREAD

**Discussion by bric3** [community/community#67891] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

The REST API to get GHA runs does not fill in `pull_requests` when a run was triggered by a pull request, i.e. the array is empty.

```
# curl -L
#  -H "Accept: application/vnd.github+json"
#  -H "Authorization: Bearer <YOUR-TOKEN>"
#  -H "X-GitHub-Api-Version: 2022-11-28"
#  https://api.github.com/repos/OWNER/REPO/actions/runs

GET https://api.github.com/repos/bric3/jufmt/actions/runs
Accept: application/vnd.github+json
Authorization: Bearer <token>
X-GitHub-Api-Version: 2022-11-28
```

E.g. this run triggered by a a pull request 
* https://github.com/bric3/jufmt/actions/runs/6223317359

The returned payload for the corresponding run (242) below, you can notice the fields `"event": "pull_request"` and that `"pull_requests": []` is empty.

```json
    {
      "id": 6223317359,
      "name": "Java CI with Gradle",
      "node_id": "WFR_kwLOEufiyM8AAAABcvBJbw",
      "head_branch": "windows-failure",
      "head_sha": "0a1e93ee775f4311bf61aa78171890c1937c7c6d",
      "path": ".github/workflows/gradle.yml",
      "display_title": "Understand windows failure",
      "run_number": 242,
      "event": "pull_request",
      "status": "completed",
      "conclusion": "success",
      "workflow_id": 20103782,
      "check_suite_id": 16285471685,
      "check_suite_node_id": "CS_kwDOEufiyM8AAAADyrCTxQ",
      "url": "https://api.github.com/repos/bric3/jufmt/actions/runs/6223317359",
      "html_url": "https://github.com/bric3/jufmt/actions/runs/6223317359",
      "pull_requests": [],
      "created_at": "2023-09-18T13:35:21Z",
      "updated_at": "2023-09-18T13:42:56Z",
      ...
```

I believe this is a bug in the API.

**Comment by Naimulnobel** [comment]
It does seem like a discrepancy in the GitHub Actions API. The `pull_requests` field should ideally contain the pull request(s) that triggered the run, especially when the `event` field is `pull_request`.

However, this seems to be a known issue with the GitHub Actions API. As per the GitHub documentation, the `pull_requests` array is not populated for runs triggered by `pull_request` events. This is a limitation of the current version of the API.

You can still get the pull request information by parsing the `html_url` or `url` field. The pull request number is included in the URL. Here's an example of how you can do this in Python:

```python
import re

# The URL of the run
run_url = "https://github.com/bric3/jufmt/actions/runs/6223317359"

# Regular expression to extract the pull request number
regex = r"https://github.com/[^/]+/[^/]+/pull/(\d+)"

# Find matches
matches = re.search(regex, run_url)

# If a match is found
if matches:
    # The first group is the pull request number
    pull_request_number = matches.group(1)
    print(f"Pull request number: {pull_request_number}")
```

This will print the pull request number if it is included in the URL.

Please note that this is a workaround and might not work if the URL structure changes in the future. It would be best if the API directly provided this information. You might want to raise this issue with GitHub's support or in the GitHub community forums.
